src/averageAges.js

```diff
@@ -1,61 +1,40 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+function avarageAge(people) {
+  const ages = people.map(person => person.died - person.born);
+
+  return ages.reduce((prev, current) => prev + current, 0) / ages.length;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+  const menCentury = men
+    .filter(person => Math.ceil(person.died / 100) === century);
+
+  return !century ? avarageAge(men) : avarageAge(menCentury); ;
```

Comment:
U just increase complexity, cuz u count and men and menCentury.. but if century === undefined.. counting `menCentury` is needless